### PR TITLE
Add validations and unit tests for models

### PR DESCRIPTION
**Pull request by utsargo**
### Description
- Add required validations and tests for the User model
- Add required validations and tests for the Post model
- Add required validations and tests for the Like model
- Add required validations and tests for the Comment model

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
```
Comment (Ismailco) [inline spec/models/user_spec.rb:3]: - [ ] All the methods in the `User` Model should be tested, kindly make sure to implement unit tests for the `most_recent_posts` method

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
```
Comment (Ismailco) [inline spec/models/post_spec.rb:3]: - [ ] All the methods in the `Post` Model should be tested, kindly make sure to implement unit tests for the `update_posts_likes_counter` method

File: spec/models/like_spec.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+require 'rails_helper'
+
+RSpec.describe Like, type: :model do
```
Comment (Ismailco) [inline spec/models/like_spec.rb:3]: - [ ] All the methods in the `Like` Model should be tested, kindly make sure to implement unit tests for the `update_posts_like_counter` method

File: spec/models/comment_spec.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
+require 'rails_helper'
+
+RSpec.describe Comment, type: :model do
```
Comment (Ismailco) [inline spec/models/comment_spec.rb:3]: - [ ] All the methods in the `Comment` Model should be tested, kindly make sure to implement unit tests for the `update_comments_counter` method

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @utsargo, 

Good job so far 👍🏼 
You still need to work on some issues to go to the next project, but you are almost there!

## Highlights

✔️ Set up Model validations
✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Owusu-Desmond — APPROVED** [review]
Hi @utsargo , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 All tests pass :100:  ✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**